frontend/packages/dev-console/src/components/pipelines/detail-page-tabs/pipeline-details/PipelineVisualization.tsx

```diff
@@ -17,21 +18,32 @@ const PipelineVisualization: React.FC<PipelineTopologyVisualizationProps> = ({
   pipelineRun,
 }) => {
   const { nodes, edges } = getTopologyNodesEdges(pipeline, pipelineRun);
-
-  if (nodes.length === 0 && edges.length === 0) {
+  let hasErrors;
+  if (hasInlineTaskSpec(pipeline)) {
+    // TODO: Inline taskSpec is not yet supported feature
+    hasErrors = (
+      <Alert
+        variant="info"
+        isInline
+        title="This Pipeline cannot be visualized. Pipeline taskSpec is not supported."
+      />
+    );
+  } else if (nodes.length === 0 && edges.length === 0) {
     // Nothing to render
     // TODO: Confirm wording with UX; ODC-1860
-    return <Alert variant="info" isInline title="This Pipeline has no tasks to visualize." />;
```

Comment:
Why did you remove the return? I think we can just have another return here.

Reply:
I have moved it inside the block, so that it gets the margin-bottom style (for the gap underneath the message) from the `odc-pipeline-visualization` class.

Reply:
Ah, I see... why don't we clean this up more and instead of treating errors differently, just have a variable called `content` or something and have 3 if-else statements... landing on rendering the Component as the fallback `else`.

It'll look cleaner and be easier to read.

Reply:
I agree, its more readable now. Updated it